### PR TITLE
Add 'no subscriptions' message to subscriptions page

### PR DESCRIPTION
**Pull request by oetherington**
Add an explanation message to the "manage subscriptions" page when the user has no subscriptions (currently the page is completely blank). Most of the diff here is just moving components into their own files. Also open to changes in copy.

EA Forum:
Screenshot 2024-08-27 at 16 12 33

LessWrong:
Screenshot 2024-08-27 at 16 13 06

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208160640868129) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/users/SubscriptionsList.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import React, { ReactNode } from "react";
```
Comment (oetherington) [inline packages/lesswrong/components/users/SubscriptionsList.tsx:1]: This is factored out of `ViewSubscriptionsList.tsx` - the only change it to add the usage of `useCountItemsContext` to keep track of the number of subscriptions that are loaded.

File: packages/lesswrong/components/users/ViewSubscriptionsPage.tsx [hunk 2]
```diff
@@ -1,272 +1,193 @@
-import React, {ReactNode} from 'react';
```
Comment (oetherington) [inline packages/lesswrong/components/users/ViewSubscriptionsPage.tsx:1]: Most of the changes here are refactoring. The only actual additions are wrapping the page in `CountItemsContextProvider` and the `NoSubscriptionsMessage` component which is shown when there are no subscriptions.

File: packages/lesswrong/components/hooks/CountItemsContext.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React, {
```
Comment (oetherington) [inline packages/lesswrong/components/hooks/CountItemsContext.tsx:1]: Each individual subscription type renders a component that loads its own list of subscriptions, which makes it difficult to know at the top-level whether or not the user has _no_ subscriptions. To handle this, we add this context which counts the number of subscriptions that are found in a ref that can be accessed at the top-level.

**Comment by oetherington** [comment]
@b0b3rt Let me know if you want a forum gate

**Comment by b0b3rt** [comment]
@oetherington I'm not sure the link to `/community` makes sense for subscribing to users (in the LW/AF context).  I don't even know if we have a link that makes sense there; I'd probably prefer a forum-gate for us having no link in that case.  Otherwise looks like an improvement, thanks!

**Review by Will-Howard — APPROVED** [review]
LGTM

**Pull request merged**